website/routes.py

```diff
@@ -58,7 +58,20 @@ def get_globals():
     user = _get_current_user()
 
     user_institutions = [{'id': inst._id, 'name': inst.name, 'logo_path': inst.logo_path_rounded_corners} for inst in user.affiliated_institutions] if user else []
-    all_institutions = [{'id': inst._id, 'name': inst.name, 'logo_path': inst.logo_path_rounded_corners} for inst in Institution.find().sort('name')]
+    all_institutions = [inst for inst in Institution.find().sort('name')]
+    dashboard_institutions = [
+        {'id': inst._id, 'name': inst.name, 'logo_path': inst.logo_path_rounded_corners}
+        for inst in all_institutions
+        if len(
+            Node.find_by_institutions(inst, query=(
```

Comment:
@caseyrollins Why does this need to go in `get_globals`. This is relatively expensive query that will happen on every request. Can this be moved to the view for the Dashboard?